contracts/src/v0.8/automation/2_0/KeeperRegistryLogic2_0.sol

```diff
@@ -62,23 +62,25 @@ contract KeeperRegistryLogic2_0 is KeeperRegistryBase2_0 {
     (bool success, bytes memory result) = upkeep.target.call{gas: s_storage.checkGasLimit}(callData);
     gasUsed = gasUsed - gasleft();
 
-    if (!success) return (false, result, UpkeepFailureReason.TARGET_CHECK_REVERTED, gasUsed, fastGasWei, linkNative);
-
-    bytes memory userPerformData;
-    (upkeepNeeded, userPerformData) = abi.decode(result, (bool, bytes));
-    if (!upkeepNeeded)
-      return (false, bytes(""), UpkeepFailureReason.UPKEEP_NOT_NEEDED, gasUsed, fastGasWei, linkNative);
-    if (userPerformData.length > s_storage.maxPerformDataSize)
-      return (false, bytes(""), UpkeepFailureReason.PERFORM_DATA_EXCEEDS_LIMIT, gasUsed, fastGasWei, linkNative);
+    if (!success) {
+      upkeepFailureReason = UpkeepFailureReason.TARGET_CHECK_REVERTED;
+    } else {
+      (upkeepNeeded, result) = abi.decode(result, (bool, bytes));
+      if (!upkeepNeeded)
+        return (false, bytes(""), UpkeepFailureReason.UPKEEP_NOT_NEEDED, gasUsed, fastGasWei, linkNative);
+      if (result.length > s_storage.maxPerformDataSize)
+        return (false, bytes(""), UpkeepFailureReason.PERFORM_DATA_EXCEEDS_LIMIT, gasUsed, fastGasWei, linkNative);
```

Comment:
should we even wrap the bytes("") just so no matter what we return the same style payload every time? So in the node we can always just unpack it and have that info.

Reply:
I like the idea - thought about doing it too - but decided against bc I wanted to keep changes to a minimum